### PR TITLE
Change visibility of returnBrokenResource and returnResource

### PR DESCRIPTION
**Pull request by marcosnils**
Following #912 we're changing the visibility of the specified methods. 

I've also found a bug in `jedis.quit()` which I'd like @nykolaslima or @HeartSaVioR to review. Please check this commit  aa9a5e2. If you think it's ok we should downmerge that commit to 2.6 and 2.7

### CODE REVIEW

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 1]
```diff
@@ -210,16 +210,6 @@ public void rename() {
   }
 
   @Test
-  public void renameOldAndNewAreTheSame() {
```
Comment (HeartSaVioR) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java]: Is it a reason to be removed?

**Comment by HeartSaVioR** [comment]
#912 again.
Could we apply same rule to JedisSentinelPool and ShardedJedisPool respectively?

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java (hunk 1 above)
Comment (marcosnils) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java]: @HeartSaVioR my bad. I wasn't running the latest unstable version of redis in my computer and the `rename` option changed a couple of days ago. 

I'll rollback this test.

**Comment by nykolaslima** [comment]
@marcosnils When we call `quit`, there is some reason to connection keep opened?

**Comment by marcosnils** [comment]
@nykolaslima no there isn't  because redis actually closes the socket. That's what my fix does, because if you issue a `quit` then Jedis will throw an IOException because the other side ended the connection.

**Comment by nykolaslima** [comment]
So it wouldn't be better if we call `disconnect` inside `quit`?

**Comment by marcosnils** [comment]
@nykolaslima that's what I did. I'm calling `client.disconnect()` inside `jedis.quit()` method. I don't understand where you're going.

**Comment by nykolaslima** [comment]
As we talked, we can't move the `disconnect` call into `BinaryClient`'s `quit` because this way we can't read the result in `BinaryJedis`.

The PR looks good to me. (Travis is falling, but Marcos said we just need to re-run it)

**Comment by HeartSaVioR** [comment]
Forgot to tell about quit. Great! Looks good to me. :)

**Comment by marcosnils** [comment]
@nykolaslima @HeartSaVioR I've already changed the visibility to SentinelPool and ShardedPool.
Please review it one last time and merge when you feel like it :)

**Comment by HeartSaVioR** [comment]
@marcosnils LGTM!
Btw, it's easier to merge sequentially, first #912, and #913.
I'll merge this after merging #912.

**Comment by HeartSaVioR** [comment]
Upmerged, and merged into master.
Thanks all!

**Comment by nykolaslima** [comment]
:+1:

**Comment by HeartSaVioR** [comment]
@Xiaoshuai Thanks for reporting! We're completely missed, and I'll apply same strategy to Pool.returnResourceObject().

**Pull request merged**